homeassistant/components/citybikes/sensor.py

```diff
@@ -116,7 +117,11 @@
         vol.Required(ATTR_NAME): cv.string,
         vol.Required(ATTR_TIMESTAMP): cv.string,
         vol.Optional(ATTR_EXTRA): vol.Schema(
-            {vol.Optional(ATTR_UID): cv.string}, extra=vol.REMOVE_EXTRA
+            {
+                vol.Optional(ATTR_UID): cv.string,
+                vol.Optional(ATTR_FREE_EBIKES): cv.positive_int,
```

Comment:
This integration is missing a library for the interface and data parsing code. We won't allow changes to this integration that involves that part of the code until it's been refactored to use a 3rd party library.

Please extract the device specific interface code and data parsing to a standalone library published on PyPI.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices